src/main/java/org/apache/hadoop/fs/PrestoFileSystemCache.java

```diff
@@ -58,6 +60,11 @@ private synchronized FileSystem getInternal(URI uri, Configuration conf, Key key
             return fs;
         }
 
+        if (map.size() > CACHE_LEAK_PROTECTION_THRESHOLD) {
```

Comment:
Make this a Hadoop config property instead, since we have access to the `Configuration` object here